routes/parking.js

```diff
@@ -1,7 +1,59 @@
 const express = require('express');
 const Parking = require('../models/parking');
+const Session = require('../models/session');
 const router = express.Router();
-const QRCode = require('qrcode')
+const QRCode = require('qrcode');
+var mongoose = require('mongoose');
+var Types = mongoose.Types,
+  ObjectId = Types.ObjectId;
+// const io = require('socket.io').listen(app)
+
+// Cached Data
+var cachedParkingLots = {}
+
+onInit();
+
+function onInit() {
+  createCache();
+}
+
+function createCache() {
+  Parking.find({}, async (err, parkingLots) => {
+
```

Comment:
why do we need caching when we have mongodb

Reply:
All calls to mongo db are async, data does not get pulled right away. It's better for responsiveness to cache the data in memory imo. Also, we're only storing session id's in the parking lot, not the session object, and fetching the session objects for each parking lot every time the parking lot data is fetched is inefficient, so I think it' better to just fetch it once and then update the cache